src/mngt/ocf_mngt_cache.c

```diff
@@ -2068,6 +2068,8 @@ void ocf_mngt_cache_stop(ocf_cache_t cache,
 
 	OCF_CHECK_NULL(cache);
 
+	env_mutex_destroy(&cache->flush_mutex);
```

Comment:
This to soon to destroy flush mutex. If stop fails for any reason, cache will be left in an invalid state. I think it would make sense to destroy flush mutex in ocf_mngt_cache_put(), in the conditional block (after ref counter == 0).

Reply:
done.